### ISSUE TITLE
policy / k8s integration: ensure that status update does not change content of CNP

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#8311]
Depending on the K8s version we are using, we update the CNP itself when updating its status - ensure that we are not accidentally updating the content of the CNP in this case as a safety check so that policy is not modified errantly by `cilium-agent`.

**Comment by aanm** [comment]
@ianvernon Can you provide more info how could this happen?

**Comment by ianvernon** [comment]
@aanm I saw a case with K8s 1.9 where a policy's status appeared to get updated by a single cilium-agent in a cluster, but that status update appeared to corrupt the policy, as the entire object was being updated in K8s. I am unable to conclusively say whether it did due to lack of visibility / logs, but there was no policy added by a user of the cluster, nor by a program / automation which adds policies to a cluster. The only event that occurred when the CNP got 'corrupted' was right after a status update from a node. I looked at the `UpdateStatus` code for a CNP when running in cases where we have to update the object itself in K8s, but could not see how this could happen. However, I was unable to come up with a better hypothesis as to how the policy got corrupted.

However, since we do not support K8s 1.9 anymore, I think we should just remove the capability of Updating objects when updating their status entirely to remove even the possibility of this happening.
